Engine/ac/global_object.cpp

```diff
@@ -213,24 +213,27 @@ void AnimateObjectEx(int obn,int loopn,int spdd,int rept, int direction, int blo
         quit("!AnimateObject: invalid object number specified");
     if (objs[obn].view < 0)
         quit("!AnimateObject: object has not been assigned a view");
-    if (loopn >= views[objs[obn].view].numLoops)
+    if (loopn < 0 || loopn >= views[objs[obn].view].numLoops)
         quit("!AnimateObject: invalid loop number specified");
+    if (sframe < 0 || sframe >= views[objs[obn].view].loops[loopn].numFrames)
+        quit("!AnimateObject: invalid starting frame number specified");
     if ((direction < 0) || (direction > 1))
         quit("!AnimateObjectEx: invalid direction");
     if ((rept < 0) || (rept > 2))
         quit("!AnimateObjectEx: invalid repeat value");
     if (views[objs[obn].view].loops[loopn].numFrames < 1)
         quit("!AnimateObject: no frames in the specified view loop");
 
-    debug_script_log("Obj %d start anim view %d loop %d, speed %d, repeat %d", obn, objs[obn].view+1, loopn, spdd, rept);
+    debug_script_log("Obj %d start anim view %d loop %d, speed %d, repeat %d, frame %d", obn, objs[obn].view+1, loopn, spdd, rept, sframe);
 
     objs[obn].cycling = rept+1 + (direction * 10);
     objs[obn].loop=loopn;
-    if (direction == 0)
-        objs[obn].frame = 0;
-    else {
-        objs[obn].frame = views[objs[obn].view].loops[loopn].numFrames - 1;
+    // reverse animation starts at the *previous frame*
+    if (direction) {
+        if (--sframe < 0)
```

Comment:
I recommend putting the decrement on another line to reduce confusion here.

Reply:
Hmm ok, I wanted to reduce number of lines :)